common/conf/paths.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <string>
+#include <common/conf.hpp>
+
+namespace mender {
+namespace common {
+namespace conf {
+namespace paths {
+
+using namespace std;
+namespace conf = mender::common::conf;
+
+string DefaultPathConfDir = conf::GetEnv("MENDER_CONF_DIR", "/etc/mender");
+string DefaultPathDataDir = conf::GetEnv("MENDER_DATA_DIR", "/usr/share/mender");
+string DefaultDataStore = conf::GetEnv("MENDER_DATASTORE_DIR", "/var/lib/mender");
+string DefaultKeyFile = "mender-agent.pem";
+
+string DefaultConfFile = DefaultPathConfDir + "/mender.conf";
```

Comment:
Ideally, yes, but we can't expose the boost functions in this file. If so you'll have to put it under `platform`.

Alternatively you could make a directory separator variable which we ifdef per platform, which is probably good enough.

Reply:
I thought we would use `boost::filesystem` on all platforms where we need to work with file system (paths).

Reply:
It makes more sense to me to add a platform-specific something like `PathJoin(string path, string suffix)` rather than just the dir separator constant. But like I said, I thought we would leave this to `boost::filesystem` everywhere where we need to work with paths. And with that, I guess all these constants should use `boost::filesystem::path` functions and operators to put together the full paths and then maybe do call `string()` on the result to get a string?

Reply:
As discussed in the standup, we will try to only return the "payload" from filesystems operations to the cross platform logic, and deal with paths and file lists and such under the hood. So should we put this file in `platform` then, perhaps?